app/Console/Commands/AdsProcessTx.php

```diff
@@ -0,0 +1,198 @@
+                    break;
+
+                case 'send_one':
+                    /** @var $sendOneTx SendOneTransaction */
+                    $sendOneTx = $transaction;
+                    $targetAddr = $sendOneTx->getTargetAddress();
+                    if ($targetAddr === $adServerAddress) {
+                        $message = $sendOneTx->getMessage();
+                        $user = User::where('uuid', hex2bin($this->extractUuidFromMessage($message)))->first();
+                        if (null === $user) {
+                            $tx->status = AdsTxIn::STATUS_RESERVED;
+                            $tx->save();
+                        } else {
+//                            $senderAddr = $sendOneTx->getSenderAddress();
+                            $amount = $sendOneTx->getAmount();
+                            // add to ledger
+                            $ul = new UserLedger;
+                            $ul->users_id = $user->id;
+                            $ul->amount = $amount;
+                            $ul->desc = $tx->txid;
+
+                            $tx->status = AdsTxIn::STATUS_USER_DEPOSIT;
+                            // tx added to ledger will not be processed again
+                            DB::transaction(function () use ($ul, $tx) {
+                                $ul->save();
+                                $tx->save();
+                            });
+                        }
+                        break;
+                    } else {
+                        $tx->status = AdsTxIn::STATUS_INVALID;
+                        $tx->save();
+                    }
+                    break;
+
+                default:
+                    $tx->status = AdsTxIn::STATUS_INVALID;
+                    $tx->save();
+                    break;
+            }
+        }
+        return self::EXIT_CODE_SUCCESS;
+    }
+
+    /**
+     * Extracts uuid from tx message.
+     * @param string $message tx message
+     * @return string uuid as hex string
+     */
+    private function extractUuidFromMessage(string $message): string
+    {
+        return substr($message, -32);
+    }
+
+    /**
+     * Updates block data.
+     * @param AdsClient $adsClient
+     * @throws CommandException
+     */
+    private function updateBlockIds(AdsClient $adsClient)
+    {
+        $attempt = 0;
+        $attemptMax = 5;
+        while (true) {
+            try {
+                $response = $adsClient->getBlockIds();
+                $updatedBlocks = $response->getUpdatedBlocks();
+                $this->info("Updated blocks: ${updatedBlocks}");
+                if ($updatedBlocks === 0) {
+                    break;
+                }
+            } catch (CommandException $exc) {
+                if (CommandError::GET_SIGNATURE_UNAVAILABLE === $exc->getCode()
+                    && ++$attempt < $attemptMax) {
+                    // try again after 3 seconds sleep
+                    sleep(3);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this call to "sleep". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS2964)